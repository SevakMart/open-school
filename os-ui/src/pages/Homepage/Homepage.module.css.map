{"version":3,"sources":["Homepage.module.scss","../../styles/Mixins/FlexMixins.scss","Homepage.module.css","../../styles/Mixins/FontMixin.scss","../../styles/Mixins/ButtonMixin.scss"],"names":[],"mappings":"AAIA;ECHE,aAAA;EACA,mBDGmB;ECFnB,iBDEwB;ECDxB,8BDC8B;EAC9B,uBAAA;AEAF;AFCE;EACE,eAAA;EACA,UAAA;EGRF,yBHSqB;EGRrB,kBHQmC;EGPnC,gBHO2C;EGN3C,eHMgD;EGLhD,qBHKsD;EACpD,cAAA;AEKJ;AFHE;ECZA,aAAA;EACA,mBDYqB;ECXrB,mBDW0B;ECV1B,8BDUkC;EAChC,UAAA;AEQJ;AFPI;EILF,yBJOM;EINN,cJOM;EINN,mBJOM;EINN,WJOM;EINN,YJOM;EINN,kBJOM;EINN,mBJOM;EINN,qBJOM;EINN,wBJOM;EINN,eAAA;EJQI,yBAAA;AEQN;AFNI;EInBF,6BJqBM;EIpBN,cJqBM;EIpBN,mBJqBM;EIpBN,WJqBM;EIpBN,YJqBM;EIpBN,kBJqBM;EIpBN,mBJqBM;EIpBN,qBJqBM;EIpBN,wBJqBM;EIpBN,eAAA;EJsBI,yBAAA;EACA,yBAAA;AEON","file":"Homepage.module.css"}