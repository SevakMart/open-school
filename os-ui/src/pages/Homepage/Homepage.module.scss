@import "../../styles/Mixins/ButtonMixin.scss";
@import "../../styles/Mixins/FlexMixins.scss";
@import "../../styles/Mixins/FontMixin.scss";

/*.categoriesMainContainer {
  background-color: #e5e5e5 !important;
  .categoriesListContainer {
    flex-wrap: wrap;
  }
}

.mentorMainContainer,
.categoriesMainContainer {
  background-color: #eeeff4;
  padding: 5% 1% 7% 1%;
  @include flexStyle(column, center, space-between);
  & h2:nth-of-type(1) {
    @include fontStyle("Montserrat", normal, 800, 2.25rem, 2.75rem);
    color: #333941;
  }
  & button:nth-of-type(1) {
    @include buttonStyle(
      transparent,
      #5e617b,
      1rem,
      none,
      4rem,
      2rem,
      2rem,
      1.375rem,
      1.375rem
    );
    border: 1px solid #5e617b;
    text-transform: uppercase;
    margin-top: 5%;
  }
  .mentorListContainer,
  .categoriesListContainer {
    width: 90%;
    @include flexStyle(row, center, space-between);
  }
}*/

.mainContainer {
  @include flexStyle(row, none, space-between);
  padding: 4% 5% 7% 6.25%;
  h2 {
    display: inline;
    width: 30%;
    @include fontStyle("Montserrat", normal, 800, 3rem, 3.688rem);
    color: #333941;
  }
  .buttonContainer {
    @include flexStyle(row, center, space-between);
    width: 35%;
    & button:nth-of-type(1) {
      @include buttonStyle(
        #5e617b,
        #ffffff,
        1rem,
        none,
        4rem,
        2rem,
        2rem,
        1.375rem,
        1.375rem
      );
      text-transform: uppercase;
    }
    & button:nth-of-type(2) {
      @include buttonStyle(
        transparent,
        #5e617b,
        1rem,
        none,
        4rem,
        2rem,
        2rem,
        1.375rem,
        1.375rem
      );
      border: 1px solid #5e617b;
      text-transform: uppercase;
    }
  }
}
