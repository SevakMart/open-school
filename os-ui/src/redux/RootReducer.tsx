import { combineReducers } from 'redux';
import chooseSubcategoryReducer from './Slices/ChoosSubcategorySlice';
import userInfoReducer from './Slices/loginUserSlice';
import allLearningPathFilterParamsReducer from './Slices/AllLearningPathFilterParamsSlice';
import forgotPasswordEmailReducer from './Slices/ForgotPasswordEmailSlice';
import signedUpUserIdReducer from './Slices/SignedUpUserIdSlice';
import suggestedCourseReducer from './Slices/SuggestedCourseSlice';
import userEnrolledCourseByCourseStatusReducer from './Slices/UserEnrolledCourseByCourseStatus';
import allLearningPathCoursesReducer from './Slices/AllLearningPathCourseSlice';
import allMentorsFilterParamsReducer from './Slices/AllMentorsFilterParamsSlice';
import allMentorsSliceReducer from './Slices/AllMentorsSlice';
import savedMentorsReducer from './Slices/SavedMentorsSlice';
import savedCourseReducer from './Slices/SavedLearningPathSlice';
import homepageMentorListReducer from './Slices/HomepageMentorSlice';
import portalStatusReducer from './Slices/PortalOpenStatus';

const rootReducer = combineReducers({
  portalStatus: portalStatusReducer,
  userInfo: userInfoReducer,
  chooseSubcategories: chooseSubcategoryReducer,
  filterParams: allLearningPathFilterParamsReducer,
  forgotPasswordEmail: forgotPasswordEmailReducer,
  signedUpUserId: signedUpUserIdReducer,
  suggestedCourse: suggestedCourseReducer,
  userEnrolledCourseByCourseStatus: userEnrolledCourseByCourseStatusReducer,
  allLearningPathCourses: allLearningPathCoursesReducer,
  allMentorsFilterParams: allMentorsFilterParamsReducer,
  allMentorsList: allMentorsSliceReducer,
  savedMentors: savedMentorsReducer,
  savedCourse: savedCourseReducer,
  homepageMentorList: homepageMentorListReducer,
});
export default rootReducer;
