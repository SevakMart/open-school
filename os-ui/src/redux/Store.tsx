import { configureStore } from '@reduxjs/toolkit';
import {
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import { createStateSyncMiddleware, initMessageListener } from 'redux-state-sync';
import rootReducer from './RootReducer';

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: {
      ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
    },
  }).concat(
    createStateSyncMiddleware({
      blacklist: ['persist/PERSIST', 'persist/REHYDRATE'],
    }),
  ),
});

initMessageListener(store);

export type RootState = ReturnType<typeof store.getState>
