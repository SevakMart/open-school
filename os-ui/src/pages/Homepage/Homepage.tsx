import { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useTranslation } from 'react-i18next';
import { RootState } from '../../redux/Store';
import HomepageHeader from '../../component/HomepageHeader/HomepageHeader';
import Footer from '../../component/Footer/Footer';
import Button from '../../component/Button/Button';
import styles from './Homepage.module.scss';
import { removeLoggedInUser } from '../../redux/Slices/loginUserSlice';
import categoriesService from '../../services/categoriesService';
import HomepageCategories from './Subcomponents/Categories/Categories';
import HomepageMentors from './Subcomponents/Mentors/Mentors';
import SignUp from '../../component/SignUp/SignUp';
import SignIn from '../../component/SignIn/SignIn';

const Homepage = () => {
  const dispatch = useDispatch();
  const userInfo = useSelector<RootState>((state) => state.userInfo);
  const [isLoggedIn, setIsLoggedIn] = useState(false);
  const [isOpen, setIsOpen] = useState(false);
  const [clickedButtonType, setClickedButtonType] = useState('');
  const { mainContainer, buttonContainer } = styles;
  const { t, i18n } = useTranslation('common');
  const handleButtonClick = (buttonType:string) => {
    switch (buttonType) {
      case 'signUp':
        setIsOpen(true);
        setClickedButtonType(buttonType);
        break;
      case 'signIn':
        setIsOpen(true);
        setClickedButtonType(buttonType);
        break;
      case 'closeButton':
        setIsOpen(false);
        break;
    }
  };

  useEffect(() => {
    let cancel = false;
    if ((userInfo as any).token) {
      categoriesService.getCategories({ page: 0, size: 6 }, (userInfo as any).token).then((res) => {
        const { status } = res;
        if (cancel) return;
        if (status === 401) {
          dispatch(removeLoggedInUser());
          setIsLoggedIn(false);
        } else setIsLoggedIn(true);
      });
    }
    // The return is to prevent memory leackage
    return () => { cancel = true; };
  }, []);

  return (
    <>
      <HomepageHeader
        handleFormVisibility={handleButtonClick}
      />
      <HomepageCategories isLoggedIn={isLoggedIn} />
      <HomepageMentors isLoggedIn={isLoggedIn} handleButtonClick={handleButtonClick} />
      <div className={mainContainer}>
        <h2>Start Your Journey Now!</h2>
        <div className={buttonContainer}>
          <Button buttonType="signUp" buttonClick={handleButtonClick}>Sign up as a Student</Button>
          <Button buttonType="signUp" buttonClick={handleButtonClick}>sign up as a mentor</Button>
        </div>
      </div>
      <Footer />
      {isOpen && clickedButtonType === 'signUp'
        ? <SignUp handleSignUpClicks={handleButtonClick} />
        : isOpen && clickedButtonType === 'signIn'
          ? <SignIn handleSignInClicks={handleButtonClick} />
          : null}
    </>
  );
};
export default Homepage;
