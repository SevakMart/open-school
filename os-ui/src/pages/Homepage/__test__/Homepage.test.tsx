import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { store } from '../../../redux/Store';
import Homepage from '../Homepage';

describe('Create several unit tests for Homepage Component', () => {
  test('Create a Snapshot test', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Homepage />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
