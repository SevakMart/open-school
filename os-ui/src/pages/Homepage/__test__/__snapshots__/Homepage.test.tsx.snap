// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create several unit tests for Homepage Component Create a Snapshot test 1`] = `
<DocumentFragment>
  <div
    class="desktopMenu"
  >
    <nav
      class="homePage"
    >
      <h2
        class="homePage_Logo"
      >
        string.homePage.navBar.logo
      </h2>
      <div
        class="homePage_leftMenu"
      >
        <p
          class="homePage_leftMenu_item"
        >
          string.homePage.navBar.exploreCategories
        </p>
        <p
          class="homePage_leftMenu_item"
        >
          string.homePage.navBar.mentors
        </p>
      </div>
      <div
        class="homePage_rightMenu"
      >
        <p
          class="homePage_rightMenu_item"
        >
          button.homePage.becomeMentor
        </p>
        <button
          class="homePage_btn"
          type="button"
        >
          button.homePage.signIn
        </button>
      </div>
    </nav>
  </div>
  <div
    class="mobileMenu"
  >
    <h2
      class="homePage_Logo"
    >
      string.homePage.navBar.logo
    </h2>
    <img
      alt="menu"
      class="mobileMenuButton "
      src="mobileMenu.svg"
    />
  </div>
  <div
    class="mainContent"
  >
    <div
      class="leftContent"
    >
      <h1>
        string.homePage.header.educationalPlatform
      </h1>
      <p>
        string.homePage.header.chooseCategories
      </p>
      <div
        class="buttonContainer"
      >
        <button
          class="headerButton headerButton__signUp"
          type="button"
        >
          button.homePage.signUp
        </button>
        <button
          class="leftContentSignInBtn"
          type="button"
        >
          button.homePage.signIn
        </button>
      </div>
    </div>
    <div
      class="rightContent"
    >
      <img
        alt="Education platform logo"
        src="HomepageMainImage.svg"
      />
    </div>
  </div>
  <div
    class="categoriesMainContainer"
  >
    <p
      class="categoriesTitle"
    >
      string.homePage.categories.title
    </p>
    <p
      class="categoriesSumtitle"
    >
      string.homePage.categories.exploreCategories
    </p>
    <div
      class="listContainer"
      style="background-color: rgb(255, 255, 255); margin: 0px 3%;"
    >
      <div
        class="gridContent"
      >
        <div
          class="mainContent"
        >
          <div
            class="loader"
          >
            <span />
            <span />
            <span />
          </div>
        </div>
      </div>
    </div>
    <div
      class="registrationButton"
    >
      <a
        href="/categories/subcategories"
      >
        <button
          class="regButton btnBelow"
          type="button"
        >
          button.homePage.seeAll
        </button>
      </a>
    </div>
  </div>
  <div
    class="mentorMainContainer"
  >
    <p
      class="MentorTitle"
    >
      string.homePage.mentors.ourMentors
    </p>
    <p
      class="MentorSubTitle"
    >
      string.homePage.mentors.learnAboutContributors
    </p>
    <div
      class="listContainer"
      style="margin: 3%;"
    >
      <div
        class="gridContent"
      >
        <div
          class="mainContent"
        >
          <div
            class="loader"
          >
            <span />
            <span />
            <span />
          </div>
        </div>
      </div>
    </div>
    <div
      class="registrationButton"
    >
      <a
        href="/categories/subcategories"
      >
        <button
          class="regButton btnAbove"
          type="button"
        >
          button.homePage.registerMentor
        </button>
      </a>
    </div>
  </div>
  <div
    class="mainContainer"
  >
    <h2>
      string.homePage.footer.startJourney
    </h2>
    <div
      class="buttonContainer"
    >
      <button
        class="signUpFooter signUpFooter__student"
        type="button"
      >
        button.homePage.signUpStudent
      </button>
      <button
        class="signUpFooter signUpFooter__mentor"
        type="button"
      >
        button.homePage.signUpMentor
      </button>
    </div>
  </div>
</DocumentFragment>
`;
